front/src/modules/users/components/UserProvider.tsx

```diff
@@ -15,7 +15,7 @@ export function UserProvider({ children }: React.PropsWithChildren) {
       setIsLoading(false);
     }
     if (data?.currentUser) {
-      setCurrentUser(data?.currentUser);
+      setCurrentUser(data?.currentUser as CurrentUser);
```

Comment:
let's avoid as CurrentUser. We should reconstruct an object if needed that has the proper type:
setCurrentUser({
...data?.currentUser,
workspaceMember: data?.currentUser.workspaceMember
})